projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html

```diff
@@ -98,15 +98,14 @@
 <ng-template #cellTemplate let-col>
         <igx-grid-cell
         class="igx-grid__td igx-grid__td--fw"
-        [class.igx-grid__td--edited]="rowID | transactionState:col.field:grid.rowEditable:grid.transactions:grid.pipeTrigger:grid.crudService.cell"
+        [class.igx-grid__td--edited]="rowID | transactionState:col.field:grid.rowEditable:grid.transactions:grid.pipeTrigger:grid.crudService.cell:grid.crudService.row"
```

Comment:
Those might be a bit overzealous for pipe triggers - each time _any_ edited cell changes the pipes on all cells run. That's preexisting though and adding the row on top will hardly make a difference I suppose.

Reply:
@damyanpetev it makes a difference as if you imagine the moment we cancel **rowEdit**, non of the cells change is triggered and once you enter edit and apply any new changes and then exit the edit row the **td--edited** class remains as nothing triggers the pipe to change its state.

Reply:
Yeah, that's your fix. My comment was about performance - because all the cells in the grid view (on all rows) are using the static crud service state as pipe argument, when you enter/exit edit on one cell, the entire view will go calculating the transaction state. But that's already tied to `crudService.cell` from before, which will trigger a lot more than row and thus your addition will likely have minimal perf impact and it should be fine :)

Reply:
o yeah, I get it. But probably at that point, there is no other way of catching that? 
Do you have smth in mind that could be beneficial for any performance improvement?